### DISCUSSION TITLE
Custom webview editor with multiple files needs to update  `localResourceRoots`

### DISCUSSION THREAD

**Discussion by paulheg** [microsoft/vscode-discussions#639] [Extension Development QnA] [answered]
Im having problems with the webview options `localResourceRoots`.
It seems to me that you can only set the webview options once and they dont get updated if I add additional roots later.

[My editor](https://github.com/paulheg/image-diff) compares two images and it is possible to set the second image later or change it.

This is problematic if the image is not in the current workspace and therefore not in the `localResourceRoots`.
This then results in the image not being accessible (HTTP 401).
I tried to update the list when the user selects the second image, but the webview does not seem to update its policy.

I would rather not add the root folder `/`.

Help appreciated.

**Comment by r3m0t** [comment, marked as answer]
Hi, make sure you're reassigning `WebView.options`. Modifying the `resourceRoots` array directly won't be picked up, as specified in the `@types/vscode`.

If that doesn't fix it then file an issue.

**Reply by paulheg** [reply]
Thank you for your answer. It seems to work.
